TerriaJS/terriajs
# Clipping box interim redesign

### What this PR does

Fixes #6836 

This PR introduces the following temporary UX fixes for clipping box:

* An option to zoom to clipping box
* An option to re-position the clipping box
* Trigger repositioning of clipping box when the user enables clipping box for the first time
* Cursor and scale point handle changes (makes it much easier to grasp)
* More robust interaction with the box

Note that this is a temporary fix which will be removed in a later re-design, so I have cut corners in a few places by accessing `private` members to get things working.

I have also tested with scene-editor to make sure the changes are compatible.

Changes:
![image](https://github.com/TerriaJS/terriajs/assets/1430/72a5581f-9f33-4f8e-b9ae-ad3ae8338a4e)

![image](https://github.com/TerriaJS/terriajs/assets/1430/8e8db390-c24a-4d52-97b4-d950cc598e81)


### Test me

- Open this [CI link](http://ci.terria.io/zoom-to-clipping-box/#share=s-8gsJuhul1IrAfitEIvu3BCeZA7Z&clean&https://gist.githubusercontent.com/na9da/737c818d6ca2ee0d22147181ae6eba0c/raw/9bd838001fd5820e65791a8e0892d29fa6005f92/melb-central-ion.json)
- Enable clipping box for the workbench item
- Observe that the reposition workflow is triggered
- Move the mouse and click on map to place it
- Navigate away from the box
- Click the `Zoom to clipping box` button to return to the clipping box
- Ensure the other features for clipping box continue to work

### Checklist

- [ ] ~There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)~ I have not added tests because this will be a temporary fix an will be overridden in the future.
